Upgrade EUI to v95.9.0

cee-chen opened the pull request:
`v95.7.0` ⏩ `v95.9.0`

> [!CAUTION]
> This PR contains the final set of Emotion conversions for all EuiForm components.
> If your plugin contains any custom CSS/styling to **EuiFormRow, EuiFormControlLayout, EuiCheckbox, EuiRadio, or EuiSwitch**, ⚠️ make sure to QA your UI to ensure no visual regressions have occurred! ⚠️ 

---

## [`v95.9.0`](https://github.com/elastic/eui/releases/v95.9.0)

- Updated `EuiSearchBar`'s optional `box.schema` prop with a new `recognizedFields` configuration. This allows specifying the phrases that will be parsed as field clauses ([#7960](https://github.com/elastic/eui/pull/7960))
- Updated `EuiIcon` with a new `tokenSemanticText` glyph ([#7971](https://github.com/elastic/eui/pull/7971))
- Added support for TypeScript 5 ([#7980](https://github.com/elastic/eui/pull/7980))

**Bug fixes**

- Fixed `EuiSelectableTemplateSitewide` styles when used within a dark-themed `EuiHeader` ([#7977](https://github.com/elastic/eui/pull/7977))

## [`v95.8.0`](https://github.com/elastic/eui/releases/v95.8.0)

- Updated `EuiHeaderLinks`'s mobile menu to set a slight popover padding by default ([#7961](https://github.com/elastic/eui/pull/7961))
  - This can be overridden via `popoverProps.panelPaddingSize` if needed.
- Updated `EuiHeaderLink` to default to a size of `s` (down from `m`) ([#7961](https://github.com/elastic/eui/pull/7961))

**Accessibility**

- Updated the `aria-label` attribute for the `EuiFieldSearch` clear button ([#7970](https://github.com/elastic/eui/pull/7970))

**Bug fixes**

- Fixed a visual bug with `<EuiDualRange showInput="inputWithPopover" />` form controls ([#7957](https://github.com/elastic/eui/pull/7957))

**Deprecations**

- Deprecated `EuiFormRow`'s `columnCompressedSwitch` display prop. Use `columnCompressed` instead, which will automatically account for child `EuiSwitch`es ([#7968](https://github.com/elastic/eui/pull/7968))
- Deprecated `EuiFormRow`'s `rowCompressed` display prop. Use `row` instead for vertical forms, or `centerCompressed` for inline forms ([#7968](https://github.com/elastic/eui/pull/7968))
- (Styling) Updated `EuiFormRow`'s `hasEmptySpaceLabel` prop to no longer attempt to automatically align its content to a vertical center. Use the `display="center"` prop for that instead ([#7968](https://github.com/elastic/eui/pull/7968))

**CSS-in-JS conversions**

- Converted `EuiFormControlLayout` to Emotion ([#7954](https://github.com/elastic/eui/pull/7954))
  - Removed `.euiFormControlLayout--*icons` classNames and `--eui-form-control-layout-icons-padding` CSS var. Use `--euiFormControlRightIconsCount` or `--euiFormControlLeftIconsCount` instead
- Converted `EuiFormLayoutDelimited` to Emotion ([#7957](https://github.com/elastic/eui/pull/7957))
- Fixed `cloneElementWithCss` throwing an error when used multiple times without a `key` prop ([#7957](https://github.com/elastic/eui/pull/7957))
- Updated `cloneElementWithCss` utility to support a third argument that allows prepending vs. appending the cloned Emotion css className ([#7957](https://github.com/elastic/eui/pull/7957))
- Removed `@euiFormControlLayoutClearIcon` Sass mixin ([#7959](https://github.com/elastic/eui/pull/7959))
- Converted `EuiDescribedFormGroup` to Emotion ([#7964](https://github.com/elastic/eui/pull/7964))
- Converted `EuiForm`, `EuiFormHelpText`, and `EuiFormErrorText` to Emotion ([#7966](https://github.com/elastic/eui/pull/7966))
- Converted `EuiFormLabel` and `EuiFormLegend` to Emotion; Removed `@euiFormLabel` mixin ([#7967](https://github.com/elastic/eui/pull/7967))
- Converted `EuiFormRow` to Emotion ([#7968](https://github.com/elastic/eui/pull/7968))
- Converted `EuiCheckbox` to Emotion ([#7969](https://github.com/elastic/eui/pull/7969))
- Converted `EuiRadio` to Emotion ([#7969](https://github.com/elastic/eui/pull/7969))
- Converted `EuiSwitch` to Emotion ([#7969](https://github.com/elastic/eui/pull/7969))
- Removed the following Sass variables: ([#7969](https://github.com/elastic/eui/pull/7969))
  - `$euiFormCustomControlDisabledIconColor`
  - `$euiFormCustomControlBorderColor`
  - `$euiRadioSize`
  - `$euiCheckBoxSize`
  - `$euiCheckboxBorderRadius`
  - `$euiSwitchHeight` (and compressed/mini variants)
  - `$euiSwitchWidth` (and compressed/mini variants)
  - `$euiSwitchThumbSize` (and compressed/mini variants)
  - `$euiSwitchIconHeight`
  - `$euiSwitchOffColor`
- Removed the following Sass mixins: ([#7969](https://github.com/elastic/eui/pull/7969))
  - `euiIconBackground`
  - `euiCustomControl`
  - `euiCustomControlSelected`
  - `euiCustomControlDisabled`
  - `euiCustomControlFocused`

cee-chen committed:
Upgrade EUI to v95.8.0

cee-chen committed:
i18n updates

cee-chen committed:
Delete/Update references to removed modifier classes

cee-chen commented:
/ci

cee-chen commented:
/ci

cee-chen commented:
/ci

cee-chen committed:
Update type from EuiFormRow class->function component migration

cee-chen committed:
Fix Enzyme tests that require more specific selectors

- caused by additional Emotion wrapper that matches the same selector

cee-chen committed:
Update Enzyme snapshots from EuiFormRow class->function component migration

- default props are no longer read/rendered by enzyme's snapshotter

cee-chen committed:
Update DOM snapshots from DOM changes/Emotion classNames

cee-chen committed:
Fix test DOM selectors to account for updated DOM

- checkbox and radio inputs should now be directly clickable, and have a different DOM wrapper

cee-chen commented:
/ci

cee-chen committed:
Fix test failures due to EuiFormRow switching from `htmlIdGenerator` to `useGeneratedHtmlId`

- requires updating mocks

cee-chen commented:
/ci

cee-chen commented:
/ci

tkajtoch commented:
buildkite test this

tkajtoch commented:
/ci

cee-chen commented:
/ci

cee-chen committed:
Replace deprecated `@euiFormLabel` mixin

- use EuiTitle size and font weight instead

cee-chen committed:
Delete unused `$euiSwitchWidth` CSS

- `.spcShareToSpaceIncludeRelated` isn't actually used anywhere in Kibana

cee-chen committed:
Replace direct className usage of several deprecated Emotion conversions with components instead

- in the cast of the `prepend` prop, `<EuiFormLabel>` is already automatically used if the prepend type is a string

cee-chen committed:
Remove direct form control append/prepend classNames

- EuiFormControlLayout should automatically handle styling, and prepend/append nodes shouldn't be used outside of form layouts

cee-chen committed:
[Unified Search] Remove direct className usage in favor of component

cee-chen committed:
Replace EuiSwitch `--mini` direct className usage with prop

cee-chen committed:
Fix CSS selector

- EuiCheckbox DOM has changed since, color: inherit should work for both disabled and non-disabled checkboxes

cee-chen commented:
/ci

cee-chen committed:
Remove unnecessary `hasEmptyLabelSpace` prop within `EuiDescribedFormGroup`s

- EuiDescribedFormGroups already automatically accounts for form rows without labels, so the `hasEmptyLabelSpace` prop is now unnecessary and adds extra unwanted spacing

cee-chen commented:
/ci

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

PhilippeOberti reviewed (approved):
changes LGTM for the Threat Hunting Investigations team

pheyos reviewed (approved):
`kbn-test-jest-helpers` change LGTM

SiddharthMantri reviewed (approved):
Tested change to Spaces plugin - Kibana security changes LGTM!

gergoabraham reviewed (approved):
@elastic/security-defend-workflows related changes look good! thanks for updating the test! 🙌 

also looked for EuiFormRow, EuiCheckbox, EuiRadio, or EuiSwitch components: there is no custom styling used for those in `x-pack/plugins/osquery` and `x-pack/plugins/security_solution/public/management` 👍

cee-chen committed:
Merge branch 'main' into eui/v95.8.0

kertal reviewed (approved):
LGTM, Codeowners review, there were just snapshot modified. I quickly checked the forms that triggered this snapshot changes, they look like expected, no Emotion regression in sight

consulthys reviewed (approved):
We went over all Stack Monitoring and alerting rules screens and they all look good.

nreese reviewed (approved):
kibana-presentation changes LGTM
code review only

cee-chen committed:
Upgrade to latest release

- contains all the backports + several requested new features from Kibana consumers

Heenawter reviewed (changes requested):
There is a pretty significant visual regression with controls on Dashboards - we have a lot of custom styling via classes, so I assume one (or more) of these styles is not being applied correctly anymore?

**On main:**

![image](https://github.com/user-attachments/assets/8bf75162-92a3-47cc-a468-50ea21089465)


**On this PR:**

![image](https://github.com/user-attachments/assets/3c4dc70d-ad29-435f-adf4-4cb6469b3701)

cee-chen committed:
Fix Dashboard controls custom CSS

- remove now-unnecessary unsets or importants

- tweak heights/vertical centering etc

- [feat] rewrite range slider CSS in Emotion vs Sass + with new CSS variables etc

cee-chen commented:
@Heenawter This should be fixed now from https://github.com/elastic/kibana/pull/190752/commits/ebce0ceb5c755f97bc0c0acce2c0f290459c02cf with some extra UI/UX pizazz added to the warning underline for the RangeSliderControl component, as we chatted about over Slack!

<img width="145" alt="" src="https://github.com/user-attachments/assets/8be775d5-fd63-4cd5-82f7-df3e8d0e32c1">
<br>
<img width="482" alt="" src="https://github.com/user-attachments/assets/f3a0ba6c-895d-4395-995f-2a7163361baf">
<br>
<img width="481" alt="" src="https://github.com/user-attachments/assets/cc3b888b-b163-4aca-8ee3-37208ca5b78d">

Let me know if you have any questions about the changes or if you spot any UI bugs that I missed.

criamico reviewed (approved):
Tested locally Fleet and Integrations and looks good! LGTM

peteharverson reviewed (commented):
For ML, spotted one regression.

mbondyra committed:
Merge branch 'main' into eui/v95.8.0

mbondyra reviewed (approved):
Visualizations team changes LGTM! Thank you!

peteharverson commented on x-pack/plugins/aiops/public/components/log_rate_analysis/item_filter_popover.tsx:
```diff
@@ -156,8 +156,7 @@ export const ItemFilterPopover: FC<ItemFilterPopoverProps> = ({
               }`}
-              className="euiSwitch--mini"
-              compressed
+              mini
```
This is on the Log rate analysis page in the AIOps Labs section of the ML plugin. Add the Kibana sample web logs data set, then run an analysis on the Log rate analysis page over the full data range and you will see the grouping control above the table.

cee-chen replied:
Should be fixed now in https://github.com/elastic/kibana/pull/190752/commits/db9659699c8c6c8e7378bd859e555dab7f07c92d!

peteharverson replied:
Thanks for fixing that!

cee-chen committed:
[PR feedback] Fix visual regression by removing unnecessary EuiFormRow

- doesn't apply and isn't necessary, removing it gets the button group back to the previous prod rendering

Heenawter reviewed (approved):
I noticed that the border on the controls is noticeably lighter in this PR than on `main`:

- **Before:**
   ![image](https://github.com/user-attachments/assets/346e16d7-7216-4e1d-96f9-03881dfffea1)

- **After:**
   ![image](https://github.com/user-attachments/assets/58c5784d-0d7c-40bb-9816-a27b6cce9d10)

Not a big deal, but is this an intentional change? 

Either way, approving to unblock, since this is pretty minor - thanks so much for your effort in cleaning up our styling, it is very, **very** much appreciated 🙇 The new range slider styling is so clean 🤌

cee-chen commented:
> I noticed that the border on the controls is noticeably lighter in this PR than on main [...] is this an intentional change?

This is an intentional change! Borders for inputs with prepend/appends (aka grouped inputs) were previously darker than inputs without them. This is now fixed so there is no change in border lightness/darkness between grouped inputs and non-grouped inputs.

peteharverson reviewed (approved):
LGTM for ML. Thanks for fixing the Smart grouping field!

Samiul-TheSoccerFan reviewed (approved):
LGTM, code review only.

Only one file has been updated `x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/components/synchronization/blocked_window_item.test.tsx`

robbavey reviewed (approved):
LGTM

cee-chen committed:
Merge branch 'main' into eui/v95.8.0

cee-chen committed:
update test from latest main

cee-chen committed:
Merge remote-tracking branch 'upstream/main' into eui/v95.8.0
